packages/flutter/test/cupertino/route_test.dart

```diff
@@ -1052,6 +1055,37 @@ void main() {
     expect(rootObserver.dialogCount, 0);
     expect(nestedObserver.dialogCount, 1);
   });
+
+  testWidgets('showCupertinoModalPopup allows for semantics dismiss on iOS', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    final SemanticsTester semantics = SemanticsTester(tester);
+    await tester.pumpWidget(CupertinoApp(
+      home: Navigator(
+        onGenerateRoute: (RouteSettings settings) {
+          return PageRouteBuilder<dynamic>(
+            pageBuilder: (BuildContext context, Animation<double> _, Animation<double> __) {
+              return GestureDetector(
+                onTap: () async {
+                  await showCupertinoModalPopup<void>(
+                    context: context,
+                    builder: (BuildContext context) => const SizedBox(),
+                  );
+                },
+                child: const Text('tap'),
+              );
+            },
+          );
+        },
+      ),
+    ));
+
+    // Push the route.
+    await tester.tap(find.text('tap'));
+    await tester.pumpAndSettle();
+
+    expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.tap], label: 'Dismiss'));
```

Comment:
nit: I'd format this as follows for slightly better readability:

```suggestion
    expect(semantics, includesNodeWith(
      actions: <SemanticsAction>[SemanticsAction.tap], 
      label: 'Dismiss',
    ));
```